tests/components/yolink/test_device_trigger.py

```diff
@@ -0,0 +1,162 @@
+            "type": "remote_button_short_press",
+            "device_id": device_entry.id,
+            "subtype": "button_2",
+            "metadata": {},
+        },
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "type": "remote_button_long_press",
+            "device_id": device_entry.id,
+            "subtype": "button_2",
+            "metadata": {},
+        },
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "type": "remote_button_short_press",
+            "device_id": device_entry.id,
+            "subtype": "button_3",
+            "metadata": {},
+        },
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "type": "remote_button_long_press",
+            "device_id": device_entry.id,
+            "subtype": "button_3",
+            "metadata": {},
+        },
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "type": "remote_button_short_press",
+            "device_id": device_entry.id,
+            "subtype": "button_4",
+            "metadata": {},
+        },
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "type": "remote_button_long_press",
+            "device_id": device_entry.id,
+            "subtype": "button_4",
+            "metadata": {},
+        },
+    ]
+    triggers = await async_get_device_automations(
+        hass, DeviceAutomationType.TRIGGER, device_entry.id
+    )
+    assert_lists_same(triggers, expected_triggers)
+
+
+async def test_if_fires_on_event(hass, calls, device_reg):
+    """Test for event triggers firing."""
+    mac_address = "12:34:56:AB:CD:EF"
+    connection = (device_registry.CONNECTION_NETWORK_MAC, mac_address)
+    config_entry = MockConfigEntry(domain=DOMAIN, data={})
+    config_entry.add_to_hass(hass)
+    device_entry = device_reg.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        connections={connection},
+        identifiers={(DOMAIN, mac_address)},
+        model=ATTR_DEVICE_SMART_REMOTE,
+    )
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: [
+                {
+                    "trigger": {
+                        "platform": "device",
+                        "domain": DOMAIN,
+                        "device_id": device_entry.id,
+                        "type": "remote_button_short_press",
+                        "subtype": "button_4",
+                    },
+                    "action": {
+                        "service": "yolink.automation",
+                        "data": {"message": "service called"},
+                    },
+                },
+            ]
+        },
+    )
+
+    device = device_reg.async_get_device(set(), {connection})
+    assert device is not None
+    # Fake remote button short press.
+    hass.bus.async_fire(
```

Comment:
Set up the integration while patching the library so we can trigger a library event that will fire a Home Assistant event. We want to access the callback that is used for the library events, `on_message_callback`. It will be inside `MagicMock.call_args`.